src/contexts/SearchContext/SearchContextProvider.tsx

```diff
@@ -0,0 +1,11 @@
+'use client';
+
+import { SearchContext, SearchContextProps } from './SearchContext';
+
+export function SearchProvider({ children }: { children: React.ReactNode }) {
+  const setSearchQuery: SearchContextProps['setSearchQuery'] = () => {};
```

Comment:
It’s hard to review what you haven’t finished, so please finish this context and fix the useOutsideClick hook and don’t add anything new to this PR. When you finish adding changes to the PR, ping me in Skype so I'll review it real fast and give you a new task